bevy_ios_safearea/src/plugin.rs

```diff
@@ -1,35 +1,78 @@
 use bevy::prelude::*;
-#[cfg(target_os = "ios")]
-use bevy::window::PrimaryWindow;
 
-#[derive(Copy, Clone, Debug, Default)]
+/// Struct providing iOS device safe area insets.
+#[derive(Resource, Clone, Debug, Default, Reflect)]
+#[reflect(Resource)]
 pub struct IosSafeArea {
+    /// The inset from the top of the screen.
+    ///
+    /// This value accounts for elements like the notch or status bar.
     pub top: f32,
+    /// The inset from the bottom of the screen.
+    ///
+    /// This value accounts for elements like the home indicator.
     pub bottom: f32,
+    /// The inset from the left side of the screen.
+    ///
+    /// This value is non-zero for devices with rounded corners or unique screen shapes.
+    pub left: f32,
+    /// The inset from the right side of the screen.
+    ///
+    /// This value is non-zero for devices with rounded corners or unique screen shapes.
+    pub right: f32,
 }
 
-#[derive(Resource, Clone, Debug, Default)]
-pub struct IosSafeAreaResource {
-    pub safe_area: Option<IosSafeArea>,
+#[allow(dead_code)]
+/// A trait providing helper methods to access the safe area insets.
+pub trait SafeAreaHelper {
```

Comment:
* why put this in a trait instead of streight `impl` on `IosSafeAreaResource` (or now `IosSafeArea`)?
* lets name consistently: `IosSafeAreaHelper`

Reply:
ok now i get it